File: _Practice/LeetCode/015. Longest Common Prefix.java
// https://leetcode.com/problems/longest-common-prefix/
// Time: O(S*log(n)), where S = sum of all characters in strs

class Solution {
    public String longestCommonPrefix(String[] strs) {
        Arrays.sort(strs);
        int SIZE = strs.length, minLen = strs[0].length(), i = 0;
        String l = strs[0], r = strs[SIZE - 1];
        for (; i < minLen; ++i) if (l.charAt(i) != r.charAt(i)) break;
        return l.substring(0, i);
    }
}

File: _Practice/LeetCode/015. Longest Common Prefix.cpp
// https://leetcode.com/problems/longest-common-prefix/
// Time: O(S*log(n)), where S = sum of all characters in strs

class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        sort(strs.begin(), strs.end());
        string l = strs[0], r = strs.back();
        int i = 0, SIZE = l.size();
        for(; i < SIZE; ++i) if(l[i] != r[i]) break;
        return l.substr(0, i);
    }
};
